subsys/bluetooth/mesh/adv_legacy.c

```diff
@@ -225,6 +229,15 @@ int bt_mesh_adv_enable(void)
 	return 0;
 }
 
+int bt_mesh_adv_disable(void)
+{
+	enabled = false;
+	k_sem_take(&adv_thread_completion_sem, K_FOREVER);
```

Comment:
I think you can use `k_thread_join` instead of the semaphore. Not sure if `k_thread_abort` is needed as the thread will be stopped after `k_thread_join` (unless it timed out).